idaes/core/initialization/general_hierarchical.py

```diff
@@ -0,0 +1,156 @@
+        ),
+    )
+    CONFIG.declare(
+        "solver_options",
+        ConfigValue(
+            default={},
+            description="Dict of options to pass to solver",
+        ),
+    )
+
+    def initialization_routine(self, model: Block):
+        """
+        Common initialization routine for models with standard form.
+
+        Args:
+            model: Pyomo Block to be initialized
+
+        Returns:
+            Pyomo solver results object
+        """
+        if not hasattr(model, "control_volume"):
+            raise TypeError(
+                f"Model {model.name} does not appear to be a standard form unit model. "
+                f"Please use an Initializer specific to the model being initialized."
+            )
+
+        # Get logger
+        _log = self.get_logger(model)
+
+        sub_initializers = {}
+        for sm in model._initialization_order:
+            if sm is not model:
+                # Get initializers for add-ons
+                sub_initializers[sm] = self.get_submodel_initializer(sm)
+
+                # Call prepare method for add-ons
+                # TODO: Need arguments for submodel initialization
+                sub_initializers[sm].prepare()
+
+        # Initialize properties
+        try:
+            # Guess a 0-D control volume
+            # Initialize inlet properties - inlet state should already be fixed
+            prop_init = self.get_submodel_initializer(
+                model.control_volume.properties_in
+            )
+
+            if prop_init is not None:
+                prop_init.initialize(
+                    solver=self.config.solver,
+                    optarg=self.config.solver_options,
+                    outlvl=self.config.output_level,
+                )
+
+            # Map solution from inlet properties to outlet properties
+            state = to_json(
+                model.control_volume.properties_in,
+                wts=StoreSpec().value(),
+                return_dict=True,
+            )
+            from_json(
+                model.control_volume.properties_out,
+                sd=state,
+                wts=StoreSpec().value(only_not_fixed=True),
+            )
+        except AttributeError:
+            # Assume it must be a 1-D control volume
+            # TODO: Add steps here
+            raise
+
+        _log.info_high("Step 1: properties initialization complete")
+
+        # Initialize reactions if they exist
+        try:
+            model.control_volume.reactions.initialize(
+                solver=self.config.solver,
+                optarg=self.config.solver_options,
+                outlvl=self.config.output_level,
+                state_vars_fixed=True,
+            )
+        except AttributeError:
+            pass
+        _log.info_high("Step 2: reactions initialization complete")
+
+        # Solve full unit model
+        solver = get_solver(self.config.solver, self.config.solver_options)
```

Comment:
As mentioned earlier, at this point initializers expect a solver name rather than a solver object. I can see argument either way for this (a solver object reduces the number of things we create and allows us to share solver objects, but a solver name keep the different sub-problems isolated and potentially allows them to use different solvers and/or options without risking cross-contamination).